cmd/src/server.rs

```diff
@@ -147,6 +156,60 @@ impl TiKVServer {
         }
     }
 
+    /// Initialize and check the config
+    ///
+    /// Warnings are logged and fatal errors exist.
+    ///
+    /// #  Fatal errors
+    ///
+    /// - If `dynamic config` feature is enabled and failed to register config to PD
+    /// - If some critical configs (like data dir) are differrent from last run
+    /// - If the config can't pass `validate()`
+    /// - If the max open file descriptor limit is not high enough to support
+    ///   the main database and the raft database.
+    fn init_config(mut config: TiKvConfig, pd_client: Arc<RpcClient>) -> ConfigController {
+        let mut version = Default::default();
+        if config.dynamic_config {
+            // Server address and cluster id can not be changed
```

Comment:
I think server address can be changed.

Reply:
Currently we use server address as id to register/get config from PD's config manager, so we should ignore change of server address on PD side in case of id conflict. /cc @overvenus

Reply:
Ignore above, we decide use advertise_addr as config id instead.